controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -708,3 +727,13 @@ func addStorageRelatedAnnotations(obj client.Object, storageConsumerName, storag
 	annotations[StorageClaimAnnotation] = storageClaim
 	annotations[StorageCephUserTypeAnnotation] = cephUserType
 }
+
+func getUnitSizeOfPgs(osdCount int) int {
+	totalPGCount := MaxPGPerOSD * osdCount
+	totalPGCount -= 16 // taking 16 pgs buffer
+	// 5 (block pool consumers) + 4 (ceph fs underlying block pool)
+	pgUnitSize := float64(totalPGCount / 9)
+	pgUnitSize = math.Min(pgUnitSize, MaxPGUnitSize)
+	nearestPowerOf2 := math.Floor(math.Log(pgUnitSize) / math.Log(2))
+	return int(math.Pow(2, nearestPowerOf2))
+}
```

Comment:
Don't have a duplicate of this function. Move it to somewhere where all code path can use it (maybe utils)